core/input/input.cpp

```diff
@@ -701,6 +701,16 @@ void Input::_parse_input_event_impl(const Ref<InputEvent> &p_event, bool p_is_em
 
 			if (!p_event->is_echo()) {
 				if (p_event->is_action_pressed(E.key)) {
+					Ref<InputEventJoypadMotion> iejm = p_event;
+					if (iejm.is_valid()) {
+						if (action.axis_pressed) {
+							// If axis is already pressed, don't increase the pressed counter.
+							action.pressed--;
+						} else {
+							action.axis_pressed = true;
+						}
+					}
+
 					action.pressed++;
```

Comment:
Decreasing and afterwards increasing `action.pressed` seems to be more difficult to understand.
What do you think about the following?
```suggestion
					Ref<InputEventJoypadMotion> iejm = p_event;
					if (iejm.is_valid()) {
						if (!action.axis_pressed) {
							action.pressed++;
							action.axis_pressed = true;
						}
					} else {
						action.pressed++;
					}
```